src/js/protocols/stm32usbdfu.js

```diff
@@ -149,17 +148,19 @@ STM32DFU_protocol.prototype.claimInterface = function (interfaceNumber) {
         // Don't perform the error check on MacOS at this time as there seems to be a bug
         // where it always reports the Chrome error "Error claiming interface." even though
         // the interface is in fact successfully claimed.
-        if (checkChromeRuntimeError() && (GUI.operating_system !== "MacOS")) {
-            console.log('Failed to claim USB device!');
+        if (checkChromeRuntimeError()) {
+            GUI.log('running cleanup, failed to claim USB interface');
+            console.log(`running cleanup Failed to claim USB interfaceNo: ${interfaceNumber} handle ${self.handle.handle}`);
             self.cleanup();
-        }
-
-        console.log(`Claimed interface: ${interfaceNumber}`);
```

Comment:
It seems illogical to say that we claimed the interface if we were unable to claim it.
Hence I put what followed into an 'else' statement, since it seemed to depend on the presence of an active interface.